File: paiza/python/C062.py
# coding: utf-8
# 自分の得意な言語で
# Let's チャレンジ！！

N = input()
cnt = 0
cool_time = 0

for i in range(int(N)):
    food = input()
    
    if cool_time == 0 and food == 'melon':
        cool_time += 10
        cnt += 1
    elif cool_time != 0:
        cool_time -=1

print(cnt)

File: paiza/php/C/C062.php
<?php
    // 自分の得意な言語で
    // Let's チャレンジ！！

    $N = trim(fgets(STDIN));
    $waiting_time = 0;
    $cnt = 0;
    
    for($i = 0; $i < $N; $i++){
        $food = trim(fgets(STDIN));
        
        if($food == 'melon' && $waiting_time == 0){
            $cnt += 1;
            $waiting_time += 10;
        }elseif($waiting_time != 0){
            $waiting_time -= 1;
        }
    }
    
    echo($cnt);
?>